### PR TITLE
Rating component

### PR DESCRIPTION
**Pull request by ChaitanyaGadodia**
- design: https://www.figma.com/file/6atJwNCeaF6vnJ4mIcHaWu/Design-System-in-Dev

### CODE REVIEW

File: packages/pebble-web/src/components/Rating.tsx [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+      this.setState({
+        stars: generateStars(maxRating, value)
+      });
+    }
+  }
+
+  setRating = (rating: number) => {
+    const { maxRating } = this.props;
+    this.setState({
+      stars: generateStars(maxRating, rating)
```
Comment (zhirzh) [inline packages/pebble-web/src/components/Rating.tsx:37]: new objects generated for all stars

File: packages/pebble-web/src/components/Rating.tsx [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    return (
+      <div className={_className}>
+        {stars.map((star, starIndex) => {
+          const rating = starIndex + 1;
+          return (
+            <span
+              key={`${name}-${rating}`}
+              onMouseEnter={() => this.setRating(rating)}
+              onMouseMove={() => this.setRating(rating)}
+              onMouseLeave={() => this.setRating(value)}
```
Comment (zhirzh) [inline packages/pebble-web/src/components/Rating.tsx:56]: why?

File: packages/pebble-web/src/components/typings/Rating.ts [hunk 3]
```diff
@@ -0,0 +1,12 @@
+export interface RatingProps {
+  name: string;
+  maxRating: number;
+  value: number;
```
Comment (zhirzh) [inline packages/pebble-web/src/components/typings/Rating.ts:4]: `value` only used to create state. rename to `initialValue` or `defaultValue`

File: packages/pebble-web/src/components/Rating.tsx (hunk 2 above)
Comment (ChaitanyaGadodia) [reply packages/pebble-web/src/components/Rating.tsx]: removed `onMouseMove`

File: packages/pebble-web/src/components/Rating.tsx [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+        {stars.map((star, starIndex) => {
+          const rating = starIndex + 1;
+          return (
+            <span
+              key={`${name}-${rating}`}
+              onMouseEnter={() => this.setRating(rating)}
+              onMouseLeave={() => this.setRating(value)}
+              onClick={() => {
+                this.setRating(rating);
+                onChange(rating);
```
Comment (zhirzh) [inline packages/pebble-web/src/components/Rating.tsx:61]: onchange is called even when component is disabled

**Review by zhirzh — APPROVED** [review]

**Pull request merged**